ballerina/soap_utils.bal

```diff
@@ -1,106 +1,198 @@
-    http:Request req = createHttpRequest(SOAP11, body, soapAction, headers);
-    do {
-        http:Response _ = check httpClient->post(path, req);
-    } on fail var err {
-        return error Error("Failed to create soap response", err);
-    }
-}
-
-function retrieveHttpClientConfig(ClientConfiguration config) returns http:ClientConfiguration {
-    return {
-        httpVersion: config.httpVersion,
-        http1Settings: config.http1Settings,
-        http2Settings: config.http2Settings,
-        timeout: config.timeout,
-        poolConfig: config?.poolConfig,
-        auth: config?.auth,
-        retryConfig: config?.retryConfig,
-        responseLimits: config.responseLimits,
-        secureSocket: config?.secureSocket,
-        circuitBreaker: config?.circuitBreaker
-    };
-}
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+import soap.wssec;
+
+import ballerina/crypto;
+import ballerina/http;
+import ballerina/mime;
+import ballerina/regex;
+
+public function validateTransportBindingPolicy(ClientConfig config) returns Error? {
+    if config.httpConfig.secureSocket is () {
+        wssec:InboundSecurityConfig|wssec:InboundSecurityConfig[] securityPolicy = config.inboundSecurity;
+        if securityPolicy is wssec:TransportBindingConfig {
+            return error Error(INVALID_PROTOCOL_ERROR);
+        } else if securityPolicy is wssec:InboundSecurityConfig[] {
+            foreach wssec:InboundSecurityConfig policy in securityPolicy {
+                if policy is wssec:TransportBindingConfig {
+                    return error Error(INVALID_PROTOCOL_ERROR);
+                }
+            }
+        }
+    }
+}
+
+public function retrieveHttpClientConfig(ClientConfig config) returns http:ClientConfiguration {
+    return {
+        httpVersion: config.httpConfig.httpVersion,
+        http1Settings: config.httpConfig.http1Settings,
+        http2Settings: config.httpConfig.http2Settings,
+        timeout: config.httpConfig.timeout,
+        poolConfig: config.httpConfig?.poolConfig,
+        auth: config.httpConfig?.auth,
+        retryConfig: config.httpConfig?.retryConfig,
+        responseLimits: config.httpConfig.responseLimits,
+        secureSocket: config.httpConfig?.secureSocket,
+        circuitBreaker: config.httpConfig?.circuitBreaker
+    };

```

Comment:
can't we do this `return config.httpConfig;` ?

Reply:
Fixed in [e9edbae](https://github.com/ballerina-platform/module-ballerina-soap/pull/103/commits/e9edbae0d08c22c3c596d9cb8a303631e3604514)